src/crafting.h

```diff
@@ -59,16 +47,21 @@ struct recipe {
         /** Second field is the multiplier */
         std::vector<std::pair<requirement_id, int>> reqs;
 
-        std::vector<byproduct> byproducts;
+        std::map<itype_id,int> byproducts;
+
         std::string cat;
         // Does the item spawn contained in container?
-        bool contained;
+        bool contained = false;
         // What does the item spawn contained in? Unset ("null") means default container.
-        itype_id container;
+        itype_id container = "null";
         std::string subcat;
-        skill_id skill_used;
+        skill_id skill_used = skill_id::NULL_ID;
```

Comment:
The same hack as in other places: wrapping it in an explicit constructor with null value.